backend/src/db/repositories/UserRepository.ts

```diff
@@ -195,6 +195,16 @@ export default class UserRepository {
         `, {user_id: userId, visited_at: dateToInsert});
     }
 
+    getLastActiveUsers(): Promise<UserRaw[]> {
+        return this.db.fetchAll(`
+            select * from users
+            where user_id in (select distinct user_id
+                              from activity_db.user_activity
+                              where visited_at > subdate(now(), interval 30 day)
+                              ) limit 1000
+        `);
```

Comment:
is this fast? Join instead of subquery would be probably faster

Reply:
Should be fast enough (the query is executed only once at server start). Join is trickier to write (need to group by all fields in the `user` table?).

Reply:
And to clarify, the improvement is from batching reads (previously users were fetched one by one, each in a separate transaction), queries are fast enough in either case.